platforms/cuda/src/CudaKernels.cpp

```diff
@@ -1650,14 +1668,29 @@ void CudaCalcNonbondedForceKernel::initialize(const System& system, const Nonbon
     // Initialize nonbonded interactions.
 
     int numParticles = force.getNumParticles();
+    // Pack the C6 coeffiecient in with sigma and epsilon, in case LJPME is being used. The C6
+    // coefficients could live in a separate array, but that would hurt cache efficiency for LJPME.
+    doLJPME = nonbondedMethod == LJPME;
     sigmaEpsilon = CudaArray::create<float2>(cu, cu.getPaddedNumAtoms(), "sigmaEpsilon");
+    if(doLJPME){
+        if (cu.getUseDoublePrecision())
+            C6s = CudaArray::create<double>(cu, cu.getPaddedNumAtoms(), "C6s");
+        else
+            C6s = CudaArray::create<float>(cu, cu.getPaddedNumAtoms(), "C6s");
+    }
     CudaArray& posq = cu.getPosq();
     vector<double4> temp(posq.getSize());
     float4* posqf = (float4*) &temp[0];
     double4* posqd = (double4*) &temp[0];
+    // The C6 coefficients for LJPME could be computed from sigma and epsilon, but it
+    // seems like a good idea to cache them and avoid many recomputations.
```

Comment:
This is mainly for the spreading / probing routines.  We'd still have to pass sigma and epsilon anyway, so I think it's worth having some kind of C6 cache.  I actually thought about going further, and making this a posq analog, so that there would be fewer cache misses in the direct space code, but went with just a plain C6 array as a first pass.